src/mpags-cipher.cpp

```diff
@@ -13,18 +15,11 @@ int main(int argc, char* argv[])
     // Convert the command-line arguments into a more easily usable form
     const std::vector<std::string> cmdLineArgs{argv, argv + argc};
 
-    // Options that might be set by the command-line arguments
-    bool helpRequested{false};
-    bool versionRequested{false};
-    std::string inputFile{""};
-    std::string outputFile{""};
-    std::string cipherKey{""};
-    bool encrypt{true};
+    ProgramSettings settings;
```

Comment:
You need to properly initialise this, otherwise you just get junk:
```suggestion
    ProgramSettings settings{ false, false, "", "", "", CipherMode::Encrypt };
```